src/core/dispatcher.ts

```diff
@@ -51,6 +52,19 @@ export class Dispatcher implements BindingObserverDelegate {
     this.application.handleError(error, `Error ${message}`, detail)
   }
 
+  private clearEventListenersForBinding(binding: Binding) {
+    const eventListener = this.fetchEventListenerForBinding(binding)
+    if (eventListener.bindings.length == 0) {
```

Comment:
Why do we need this check `eventListener.bindings.length == 0`? In which scenario can this be invoked where this doesn't validate? I don't see anything wrong I just want to understand 😅.

Reply:
Please ask! 
It's a rough edge but It's possible to have multiple bindings for the same `EventListener`. Let's say we have this scenario:

```
<div data-controller="boosts">
  <button data-action="click->boosts#doSomething click->boosts#doSomething">Hey!</button>
</div>
```

In this case, if change the "data-action" we don't want to disconnect the `EventListener` unless we remove both "click->boosts#doSomething" actions.
Right now we call this method only if the whole controller is removed so the safeguard is not a necessity, it was a necessity in the first version I made which didn't use the "clearEventListeners" flag.